packages/indexer/native/src/lib.rs

```diff
@@ -456,11 +532,18 @@ declare_types! {
                 end_key.extend_from_slice(&u64::MAX.to_be_bytes());
             }
 
+            let skip = cx.argument::<JsValue>(6)?;
+            let skip_number = if skip.is_a::<JsNumber>() {
+                skip.downcast::<JsNumber>().or_throw(&mut cx)?.value() as usize
+            } else {
+                0_usize
+            };
+
             let iter = store.iter(&start_key, IteratorDirection::Forward);
             if iter.is_err() {
                 return cx.throw_error("Error creating iterator!");
             }
-            let iter = iter.unwrap().take_while(move |(key, _)| key.to_vec() < end_key);
+            let iter = iter.unwrap().take_while(move |(key, _)| key.to_vec() < end_key).filter(move |(key, _)| key.ends_with(&io_type_mark)).skip(skip_number);
```

Comment:
Same performance quirk with the above.

Reply:
It's slightly different from the `JsLiveCellIterator` case, cause the `io_type` is not sequencially store in rocksdb, so the one by one comparison seems unavoidable here.

Reply:
You are right, ignore this comment here.